app/src/main/java/org/schabi/newpipe/player/helper/PlayerHelper.java

```diff
@@ -162,14 +169,46 @@ public static String resizeTypeOf(@NonNull final Context context,
 
     @NonNull
     public static String cacheKeyOf(@NonNull final StreamInfo info,
-                                    @NonNull final VideoStream video) {
-        return info.getUrl() + video.getResolution() + video.getFormat().getName();
+                                    @NonNull final VideoStream videoStream) {
+        String cacheKey = info.getUrl() + " " + videoStream.getId();
+
+        final String resolution = videoStream.getResolution();
+        final MediaFormat mediaFormat = videoStream.getFormat();
+        if (resolution.equals(RESOLUTION_UNKNOWN) && mediaFormat == null) {
+            final String content = videoStream.getContent();
+            cacheKey += " " + content.substring(20 * content.length() / 100);
```

Comment:
I'm not so sure about this way of creating a key: the first 20% of the content might be equal for some streams, wouldn't it? E.g. for videos with normal urls, the first 20% characters might always be the same. What about using the hash of the string? It wouldn't ensure uniqueness, but the probability of clashes should be pretty low.

Reply:
I didn't have an idea to how use the content, but your idea of using the hash code is really great! Why not using the hash code of the stream object itself instead of using the hashcode of the content?

Reply:
Yeah, that makes sense, though the `Stream` class does not seem to have `hashCode` overridden, and the default implementation obviously [does not return something useful](https://stackoverflow.com/questions/2237720/what-is-an-objects-hash-code-if-hashcode-is-not-overridden). So either you make changes in the extractor and override the `hashCode()` method in the various `*Stream` classes, or you just use `getContent().hashCode()` and call it a day. (obviously only use hashes if `resolution.equals(RESOLUTION_UNKNOWN) && mediaFormat == null`)

Reply:
> Override the hashCode() method in the various *Stream classes

What do you suggest me to use to generate the hash code?

Reply:
See https://github.com/TeamNewPipe/NewPipeExtractor/pull/663#discussion_r790264754